packages/cli/lib/services/compile.service.ts

```diff
@@ -86,43 +87,83 @@ export async function compileAllFiles({
     return success;
 }
 
-export function compileSingleFile({
+export async function compileSingleFile({
     file,
-    tsconfig,
     config,
-    modelNames
+    modelNames,
+    tsconfig,
+    debug = false
 }: {
     file: ListedFile;
     tsconfig: string;
     config: StandardNangoConfig[];
     modelNames: string[];
+    debug: boolean;
 }) {
-    // This needs to be re-declared each time
-    const compiler = tsNode.create({
-        skipProject: true, // when installed locally we don't want ts-node to pick up the package tsconfig.json file
-        compilerOptions: JSON.parse(tsconfig).compilerOptions
-    });
-
     try {
-        compile({ file, config, compiler, modelNames });
+        const compiler = tsNode.create({
+            skipProject: true, // when installed locally we don't want ts-node to pick up the package tsconfig.json file
+            compilerOptions: JSON.parse(tsconfig).compilerOptions
+        });
+
+        return compile({ file, config, modelNames, compiler, debug });
     } catch (error) {
         console.error(`Error compiling ${file.inputPath}:`);
         console.error(error);
-        return;
+        return false;
+    }
+}
+
+function compileImportedFile(filePath: string, compiler: tsNode.Service, type: SyncConfigType | undefined, modelNames: string[]): boolean {
+    let finalResult = true;
+    const importedFiles = parserService.getImportedFiles(filePath);
+
+    if (!parserService.callsAreUsedCorrectly(filePath, type, modelNames)) {
+        return false;
+    }
+
+    for (const importedFile of importedFiles) {
+        const importedFilePath = path.resolve(path.dirname(filePath), importedFile);
+        const importedFilePathWithExtension = importedFilePath + '.ts';
+
+        /// if it is a library import then we can skip it
+        if (!fs.existsSync(importedFilePathWithExtension)) {
+            continue;
+        }
+
+        const cwd = process.cwd().split('/').pop();
+        const fullPath = path.resolve(importedFilePathWithExtension);
+        if (cwd === 'nango-integrations' && fullPath.includes(process.cwd())) {
```

Comment:
shouldn't it be the negative? ie: if fullPath doesn't includes nango-integration then show error?

Reply:
Thanks, this was the last thing I added last night without testing thoroughly. Updated and added a test case to support it + improved the error message